packages/material-ui/src/InputBase/InputBase.test.js

```diff
@@ -216,11 +216,9 @@ describe('<InputBase />', () => {
     }
 
     it('should check that the component is uncontrolled', () => {
-      const { wrapper, handleFilled } = setup();
-      assert.strictEqual(handleFilled.callCount, 1);
-      wrapper.setProps({ defaultValue: 'stub' });
-      assert.strictEqual(wrapper.find('input').prop('defaultValue'), 'stub');
-      assert.strictEqual(handleFilled.callCount, 2);
+      const { wrapper } = setup();
+      wrapper.find('input').instance().value = 'stub';
+      assert.strictEqual(wrapper.find('input').instance().value, 'stub');
```

Comment:
This is asserting what you did in the first line. Could you just dispatch a input change event?

Reply:
I’m not sure what benefit this test has realistically. Testing that checkDirty is called when uncontrolled  and isn't called when controlled seems like a better test

Reply:
To be honest I don't even know why we checkDirty only once when uncontrolled. Not sure what the use case is. Maybe a better test could help here.

Reply:
I’m not sure why either.

Reply:
For the test 'should check that the component is uncontrolled' would it be okay if i did something like this would that be okay 💭 

1- Pass in a default value to the component
2- Then use an onChange (spy) method for trying to change the input (which won't work)
3- Assert that default value remains intact & default value is the same as the input.instance().value

(absolutely not sure about this approach =P) 😅

Reply:
Hmm, I'm not sure either. Give it a try, it will be easier once we see the test

Reply:
I have this test

```js
it('should check that the component is uncontrolled', () => {
  const { wrapper } = setup();
  wrapper.setProps({ defaultValue: 'stub' });
  const handleChange = spy();
  wrapper.setProps({
    onChange: handleChange
  });
  wrapper.update();
  const event = {
    target: {
      value: 'foo'
    }
  };
  wrapper.find('input').simulate('change', event);
  assert.strictEqual(handleChange.callCount, 1);
  const inputEl = wrapper.find('input');
  console.log(inputEl.debug());
  console.log(inputEl.props());
  console.log(inputEl.html());
  assert.strictEqual(wrapper.find('input').instance().value, 'stub'); // THIS FAILS 😞 
});
```

Response for `console.log(inputEl.debug());`

```diff
<input aria-invalid={[undefined]} aria-describedby={[undefined]} autoComplete={[undefined]} autoFocus={[undefined]} className="" 
+ defaultValue="stub" 
disabled={[undefined]} id={[undefined]} name={[undefined]} onBlur={[Function: handleBlur]} onChange={[Function: handleChange]} onFocus={[Func
tion: handleFocus]} onKeyDown={[undefined]} onKeyUp={[undefined]} placeholder={[undefined]} readOnly={[undefined]} required={[undefined]} rows={[undefined]}
 value={[undefined]} type="text" />
```

Response for `console.log(inputEl.props())`

```
{ 
  'aria-invalid': undefined,
  'aria-describedby': undefined,
  autoComplete: undefined,
  autoFocus: undefined,
  className: '',
  defaultValue: 'stub',
  disabled: undefined,
  id: undefined,
  name: undefined,
  onBlur: [Function: handleBlur],
  onChange: [Function: handleChange],
  onFocus: [Function: handleFocus],
  onKeyDown: undefined,
  onKeyUp: undefined,
  placeholder: undefined,
  readOnly: undefined,
  required: undefined,
  rows: undefined,
  value: undefined,
  type: 'text' 
}
```

But the response for 'console.log(inputEl.html());` is 😭 😥 

```html
<input class="" type="text" value="hell">
```

Is my understanding of the testing wrong, shouldn't this be `assert.strictEqual(wrapper.find('input').instance().value, 'stub');` but it is `false`

Reply:
For some reason your defaultValue isn’t being set through setProps. I can have a look later but I don’t think you need wrapper.update there?

Reply:
This won't really test a lot since the value is set outside the InputBase and not in onChange. Your test would only check that defaultValue sets value and onChange is called. My guess why the current test doesn't work is due to jsdom and it's weird input `value` updating.

Reply:
I am sorry, but I can think of a way to test this case... 😞 😞

Reply:
I've removed the check state tests. I don't think they provide any value, all we care is that the component works in an uncontrolled and controlled state, which the other tests already test.